plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -298,7 +296,13 @@ public static Object convert(Object value, Type type)
             return ((Long) value) * PICOSECONDS_PER_MICROSECOND;
         }
         if (type instanceof Types.FloatType) {
-            return Float.floatToIntBits((Float) value);
+            return ((Float) value).longValue();
+        }
+        if (type instanceof Types.IntegerType || type instanceof Types.DateType) {
+            return Long.parseLong(value.toString());
+        }
+        if (type instanceof Types.DecimalType) {
```

Comment:
you should distinguish between short decimal (precision <= 18) and long decimals. They have different representation. Please add a test case for long decimal too